### PR TITLE
mysql: add mysql password option

### PR DESCRIPTION
**Pull request by shyim**
I need for the https://www.adminer.org a password to access the database.

### CODE REVIEW

File: src/modules/mysql.nix [hunk 1]
```diff
@@ -47,6 +47,18 @@ let
       fi
     '') cfg.initialDatabases}
 
+    ${optionalString (cfg.rootPassword != null) ''
+      # Set root password
+
+      if [[ ! -f "$MYSQL_HOME/.root_password_set" ]]; then
+        ${cfg.package}/bin/mysql ${mysqlOptions} -u root -N -e "ALTER USER 'root'@'localhost' IDENTIFIED BY '${cfg.rootPassword}';"
+
+        touch "$MYSQL_HOME/.root_password_set"
```
Comment (domenkozar) [inline src/modules/mysql.nix:56]: This should include the password hash so if it changes, the password is changes

Comment (shyim) [reply src/modules/mysql.nix:56]: You need the old password to set the new one

Comment (shyim) [reply src/modules/mysql.nix:56]: Maybe we allow adding additional users using options and let the `root` be root and only accessible using socket?

**Comment by shyim** [comment]
I will rework this to allow adding users

**Pull request closed without merging**